src/doctors/doctors.service.ts

```diff
@@ -1,33 +1,27 @@
 import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Specialization } from 'src/specializations/entities/specialization.entity';
-import { User } from 'src/users/entities/user.entity';
 import { Repository } from 'typeorm';
+
 import { CreateDoctorInput } from './dto/create-doctor.input';
 import { Doctor } from './entities/doctor.entity';
+import { UsersService } from 'src/users/users.service';
+import { SpecializationsService } from 'src/specializations/specializations.service';
 
 @Injectable()
 export class DoctorsService {
   constructor(
+    private readonly usersService: UsersService,
+    private readonly specializationsService: SpecializationsService,
     @InjectRepository(Doctor)
     private doctorsRepository: Repository<Doctor>,
-    @InjectRepository(User)
-    private usersRepository: Repository<User>,
-    @InjectRepository(Specialization)
-    private specializationsRepository: Repository<Specialization>,
   ) {}
+
   async create(createDoctorInput: CreateDoctorInput) {
     const doctor = await this.doctorsRepository.create(createDoctorInput);
-    console.log(doctor);
-
-    const spec = await this.specializationsRepository.findOne({
-      id: createDoctorInput.specializationId,
-    });
-    const user = await this.usersRepository.findOne({
-      id: createDoctorInput.userId,
-    });
-    console.log(spec);
-
+    const spec = await this.specializationsService.findOneById(
+      createDoctorInput.specializationId,
+    );
+    const user = await this.usersService.findOneById(createDoctorInput.userId);
     doctor.specialization = spec;
     doctor.user = user;
```

Comment:
You can just set the id columns to create a relation. (Not necessary to query the entities spec and user, as you don't check for existence or so ever).